src/Features/Core/Portable/CodeFixes/AddExplicitCast/AbstractAddExplicitCastCodeFixProvider.cs

```diff
@@ -0,0 +1,399 @@
+                    && parameterType is IArrayTypeSymbol paramsType
+                    && ClassifyConversionExists(semanticModel, argumentExpression, paramsType.ElementType))
+                {
+                    newArguments.Add(GenerateNewArgument(arguments[i], paramsType.ElementType));
+                    FindTargetArgumentAndConversionType(semanticModel, arguments[i], argumentExpression,
+                        paramsType.ElementType, ref targetArgument, ref targetArgumentConversionType);
+                }
+                else if (ClassifyConversionExists(semanticModel, argumentExpression, parameterType))
+                {
+                    newArguments.Add(GenerateNewArgument(arguments[i], parameterType));
+                    FindTargetArgumentAndConversionType(semanticModel, arguments[i], argumentExpression, parameterType,
+                        ref targetArgument, ref targetArgumentConversionType);
+                }
+                else if (IsDeclarationExpression(argumentExpression)
+                    && semanticModel.GetTypeInfo(argumentExpression, cancellationToken).Type is ITypeSymbol argumentType
+                    && semanticModel.Compilation.ClassifyCommonConversion(argumentType, parameterType).IsIdentity)
+                {
+                    // Direct conversion from a declaration expression to a type is unspecified, thus we classify the
+                    // conversion from the type of declaration expression to the parameter type
+                    // An example for this case:
+                    // void Foo(out int i) { i = 1; }
+                    // Foo([|out var i|]);
+                    // "var i" is a declaration expression
+                    // 
+                    // In addition, since this case is with keyword "out", the type of declaration expression and the
+                    // parameter type must be identical in order to match.
+                    newArguments.Add(arguments[i]);
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            return targetArgumentConversionType != null && targetArgument != null
+                && IsInvocationExpressionWithNewArgumentsApplicable(semanticModel, root, argumentList, newArguments, targetArgument);
+
+            void FindTargetArgumentAndConversionType(SemanticModel semanticModel, SyntaxNode argument,
+                TExpressionSyntax argumentExpression, ITypeSymbol conversionType, ref SyntaxNode? targetArgument,
+                ref ITypeSymbol? targetArgumentConversionType)
+            {
+                if (targetArgument is null && !IsConversionIdentity(semanticModel, argumentExpression, conversionType))
+                    targetArgument = argument;
+                if (argument.Equals(targetArgument))
+                    targetArgumentConversionType = conversionType;
+            }
+        }
```

Comment:
did this method change, or just move up?

Reply:
This method is just to extract the duplicate code:)